metron-platform/metron-parsing/metron-parsers-common/src/main/java/org/apache/metron/parsers/ParserRunnerImpl.java

```diff
@@ -256,7 +257,9 @@ private void initializeParsers(Supplier<ParserConfigurations> parserConfigSuppli
     if (!message.containsKey(Constants.GUID)) {
       message.put(Constants.GUID, UUID.randomUUID().toString());
     }
-    message.putIfAbsent(Fields.ORIGINAL.getName(), new String(rawMessage.getMessage(), StandardCharsets.UTF_8));
+    message.putIfAbsent(Fields.ORIGINAL.getName(),
```

Comment:
Do you need to actually add the method? You already have the parser available, and it has the `getReadCharset()` method, which has already done that work from the `configure()` methods?  Unless there's a call ordering issue I'm not recalling, or something.

Reply:
Gah, good catch - I forgot I added getReadCharset() to the interface

Reply:
Fixed that to use the interface method now